components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
 }
 
 - (void)commonBannerViewInit {
-  self.backgroundColor = [UIColor whiteColor];
+  self.backgroundColor = MDCColorFromRGB(MDCBannerDefaultBackgroundColor);
+  _containerView = [[UIView alloc] initWithFrame:CGRectZero];
+  _containerView.translatesAutoresizingMaskIntoConstraints = NO;
+  [self addSubview:_containerView];
 
-  _buttons = [[NSArray alloc] init];
-  self.numberOfButtons = 1;
+  _buttons = [[NSMutableArray alloc] init];
 
   UILabel *textLabel = [[UILabel alloc] init];
-  textLabel.font = [UIFont systemFontOfSize:kTextFontSize];
-  textLabel.textColor = [UIColor colorWithWhite:0 alpha:kTextColorOpacity];
+  textLabel.translatesAutoresizingMaskIntoConstraints = NO;
+  textLabel.font = [MDCTypography body2Font];
+  textLabel.textColor = UIColor.blackColor;
+  textLabel.alpha = [MDCTypography body2FontOpacity];
   textLabel.numberOfLines = kTextNumberOfLineLimit;
   textLabel.lineBreakMode = NSLineBreakByTruncatingTail;
   _textLabel = textLabel;
-  [self addSubview:textLabel];
+
+  _isLayoutDirty = YES;
+
+  [self.containerView addSubview:textLabel];
 }
 
 #pragma mark - Property Setter and Getter
 
 - (void)setText:(NSString *)text {
   self.textLabel.text = text;
+  self.isLayoutDirty = YES;
```

Comment:
Tip: `[self setNeedsLayout]`

Reply:
Does `setNeedsLayout` ensure an recalculation to `intrinsicContentSize`?

Reply:
I think I forgot to update this comment, I believe you'll need to call 
```
[self invalidateIntrinsicContentSize];
[self setNeedsLayout];
```

Reply:
Followed up offline.